Demo/Application/Features/ShopperInsightsViewController.swift

```diff
@@ -95,25 +95,50 @@ class ShopperInsightsViewController: PaymentButtonBaseViewController {
                 let result = try await shopperInsightsClient.getRecommendedPaymentMethods(request: request, experiment: sampleExperiment)
                 // swiftlint:disable:next line_length
                 progressBlock("PayPal Recommended: \(result.isPayPalRecommended)\nVenmo Recommended: \(result.isVenmoRecommended)\nEligible in PayPal Network: \(result.isEligibleInPayPalNetwork)")
-                payPalVaultButton.isEnabled = result.isPayPalRecommended
-                venmoButton.isEnabled = result.isVenmoRecommended
+                
+                togglePayPalVaultButton(enabled: result.isPayPalRecommended)
+                toggleVenmoButton(enabled: result.isVenmoRecommended)
             } catch {
                 progressBlock("Error: \(error.localizedDescription)")
             }
         }
     }
     
+    private func togglePayPalVaultButton(enabled: Bool) {
+        payPalVaultButton.isEnabled = enabled
+        
+        guard enabled else { return }
+        
+        let presentmentDetails = BTPresentmentDetails(
+            buttonOrder: .first,
+            experimentType: .control,
+            pageType: .about
+        )
+        
+        shopperInsightsClient.sendPresentedEvent(
+            for: .payPal,
+            presentmentDetails: presentmentDetails
+        )
+    }
+    
+    private func toggleVenmoButton(enabled: Bool) {
+        venmoButton.isEnabled = enabled
+        
+        guard enabled else { return }
+        
+        let presentmentDetails = BTPresentmentDetails(
+            buttonOrder: .second,
+            experimentType: .control,
+            pageType: .about
+        )
+        
+        shopperInsightsClient.sendPresentedEvent(
+            for: .venmo,
+            presentmentDetails: presentmentDetails
+        )
+    }
```

Comment:
Take it or leave it but we could combine this logic if we wanted:
```suggestion
    private func toggleButtons(result: BTShopperInsightsResult) {
        payPalVaultButton.isEnabled = result.isPayPalRecommended
        venmoButton.isEnabled = result.isVenmoRecommended

        let payPalPresentmentDetails = BTPresentmentDetails(
            buttonOrder: .first,
            experimentType: .control,
            pageType: .about
        )

        let venmoPresentmentDetails = BTPresentmentDetails(
            buttonOrder: .second,
            experimentType: .control,
            pageType: .about
        )

        if result.isPayPalRecommended {
            shopperInsightsClient.sendPresentedEvent(
                for: .payPal,
                presentmentDetails: payPalPresentmentDetails
            )
        }

        if result.isVenmoRecommended {
            shopperInsightsClient.sendPresentedEvent(
                for: .venmo,
                presentmentDetails: venmoPresentmentDetails
            )
        }
    }
```